### DISCUSSION TITLE
Plans for `starry` development

### DISCUSSION THREAD

**Discussion by dfm** [exoplanet-dev/jaxoplanet#23] [Ideas]
This thread can be a place to brainstorm plans for getting sufficient support of `starry`-like features into JAX. I've also started [a project board to track specific tasks](https://github.com/orgs/exoplanet-dev/projects/3), but this is a good place to brainstorm things at a higher level.

**Comment by shashankdholakia** [comment]
It's great to see the core functionality of starry being implemented into jaxoplanet and I'm excited to see how this goes! I wonder what is the planned scope of starry-like features in jaxoplanet. I can think of tons of useful features of starry (in no particular order):

- A starry-like API with a Primary and Secondary objects in a System (using KeplerianOrbit internally)

- Several kinds of maps:
    - Reflected light maps
    - Radial velocity maps
    - Doppler maps
    - Oblate maps (already addressed in #20)
    - multiwavelength maps
    - Perhaps a nice API for loading in maps from images (like the starry map.load using a spherical harmonic transform)
  
Of these, I think the first point is something I'd be most excited to see. Are these features all within the scope of future jaxoplanet?

**Reply by ssagynbayeva** [reply]
Would it be possible to create light curves with jaxoplanet the way starry does? I assume that jaxoplanet would also use the starry algorithm for limb darkening (?) -- so, as Shashank mentioned, it would be awesome to combine KeplerianOrbit with the starry Kepler algorithm.

**Reply by dfm** [reply]
@shashankdholakia — Thanks for this list! I think this is all within scope, but it depends on everyone's motivation/capacity (my motivation is high, but capacity is extremely limited) to get to it. Some comments:

> A starry-like API with a Primary and Secondary objects in a System (using KeplerianOrbit internally)

This is nominally already approximately supported by the existing orbit interfaces such that it shouldn't be too hard to get it running once we have a working implementation of all the `starry` matrices.

> Several kinds of maps

This is great - can you open a new issue for each of these with a little bit more info (paper and code references, for example) so that we can track development more easily.

> Perhaps a nice API for loading in maps from images (like the starry map.load using a spherical harmonic transform)

I don't think there's much that's `starry`-specific in the implementation in starry so that shouldn't be too hard to implement - open an issue or PR!

@ssagynbayeva — The goal isn't necessarily for `jaxoplanet` to reach feature parity with `starry`, but the hope is to be able to produce `starry`-like light curves in the future. We're not there yet though so using `starry` directly is still probably the right call for any research projects.

**Comment by ssagynbayeva** [comment]
I just want to emphasize the importance of additional `starry` features like [`starry_process`](https://starry-process.readthedocs.io/en/latest/) (which is its own separate package but connected to starry). It makes it easier to analyze surfaces of stars using multiple light curves with a tractable Gaussian Process kernel (also makes it super easy to create a simulated light curves sampled from GP)! Though JAXifying `starry_process` might not make the sampling faster, but I am sure there’s some workaround (i.e. some re-parametrization) that could help the problem! Rodrigo has a great [post on it](https://luger.dev/starry-process-about.html). I know it’s not an urgent goal, but I felt it’s important to point this out!

**Reply by shashankdholakia** [reply]
Seconded! Interpretable GPs with spherical harmonics would be very useful for modeling starspots and surface features and adding to the starry/jaxoplanet ecosystem!
